### PR TITLE
Node crash after moving to iotex-election 0.1.10 #1194

### PR DESCRIPTION
**Pull request by lzxm160**
work on #1194

According to #1210 (review)

First fixs poll protocol initializes.

Tests in my local computer shows that wait for about 6 minutes util initialize ok

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -311,6 +312,9 @@ type (
 
 		// SQLite3 is the config for SQLITE3
 		SQLITE3 SQLITE3 `yaml:"SQLITE3"`
+
+		// InitTryInterval is the config for committee init db
+		InitTryInterval int `yaml:"initTryInterval"`
```
Comment (zjshen14) [inline config/config.go]: Move the config to poll, and try to default to 3sec

File: action/protocol/poll/protocol.go [hunk 2]
```diff
@@ -194,9 +198,26 @@ func (p *governanceChainCommitteeProtocol) Initialize(
...
+	interval, ok := InitTryInterval.(int)
+	if !ok {
+		log.L().Error("Interval read from config error")
+		interval = 15
+	}
 	log.L().Info("Initialize poll protocol", zap.Uint64("height", p.initGravityChainHeight))
 	var ds state.CandidateList
 	if ds, err = p.delegatesByGravityChainHeight(p.initGravityChainHeight); err != nil {
+		errMsg := "bucket = electionNS doesn't exist: not exist in DB"
+		for strings.Contains(err.Error(), errMsg) {
```
Comment (zjshen14) [inline action/protocol/poll/protocol.go]: errors.Cause()

File: action/protocol/poll/protocol.go [hunk 3]
```diff
@@ -45,6 +46,9 @@ type GetEpochHeight func(uint64) uint64
 // GetEpochNum defines a function to get epoch number given a block height
 type GetEpochNum func(uint64) uint64
 
+// InitTryIntervalCtxKey is for InitTryInterval context
+type InitTryIntervalCtxKey struct{}
```
Comment (zjshen14) [inline action/protocol/poll/protocol.go]: You can set the interval when constructing the protocol

No need to pass it over context

File: config/config.go (hunk 1 above)
Comment (lzxm160) [reply config/config.go]: got it

File: action/protocol/poll/protocol.go (hunk 3 above)
Comment (lzxm160) [reply action/protocol/poll/protocol.go]: sure

**Comment by lzxm160** [comment]
> Would you please test it against mainnet (sync from a clean state) after applying this change?

already tested

File: action/protocol/poll/protocol.go [hunk 4]
```diff
@@ -197,6 +201,13 @@ func (p *governanceChainCommitteeProtocol) Initialize(
 	log.L().Info("Initialize poll protocol", zap.Uint64("height", p.initGravityChainHeight))
 	var ds state.CandidateList
 	if ds, err = p.delegatesByGravityChainHeight(p.initGravityChainHeight); err != nil {
+		for err != nil && errors.Cause(err) == db.ErrNotExist {
```
Comment (zjshen14) [inline action/protocol/poll/protocol.go]: L203-L209, how about the following?

```suggestion
    var err error
		for {
			log.L().Error("calling committee,waiting for a while", zap.Int64("duration", int64(p.initialCandidatesInterval.Seconds())), zap.String("unit", " seconds"))
			time.Sleep(p.initialCandidatesInterval)
			ds, err = p.delegatesByGravityChainHeight(p.initGravityChainHeight)
			if err == nil || errors.Cause(err) != db.ErrNotExist {
			    break
			}
		}
```

**Pull request merged**